SuperGoodSoft/solidus_taxjar
# Manual retry of failed transaction syncs - [Part 1] The Refactor.

What is the goal of this PR?
---
- Allow feature specs to run without `--headless` mode
- Add failing feature spec for retrying failed transaction syncs
- Introduce a reportable concern to encapsulate reporting criteria

This work is to address #224. We've started extracting the logic for when to report or replace a transaction in order to achieve the end goal.

**⚠️ NOTE:** We are looking for suggestions of what to name this concern as the reporting subscriber or the controller/views where we will be using this are not necessarily a `Reportable`, but we couldn't come up with anything better.

How do you manually test these changes? (if applicable)
---
With the Reporting feature enabled: 
1. Ship an order
    * [x] It should be reported to TaxJar
2. Update an order (w/ refund or additional payment)
    * [x] The transaction should be replaced on TaxJar
3. Update an order completed before reporting was enabled
    * [x] An error transaction log should be created
4. Ship an order completed before reporting was enabled
    * [x] An error transaction log should be created
    
With the Reporting feature disabled:
1. Update an order (w/ refund or additional payment)
    * [x] No transaction should be created on TaxJar
    * [x] No error transaction log should be created
    
Merge Checklist
---


Screenshots
---
N/A
